modules/custom/openy_font/openy_font.module

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @file
+ * Contains openy_font.module.
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\file\FileInterface;
+
+/**
+ * Implements hook_help().
+ */
+function openy_font_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the openy_font module.
+    case 'help.page.openy_font':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('OpenY Font settings module.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Checks that the filename is allowed.
+ *
+ * @param \Drupal\file\FileInterface $file
+ *   A file entity.
+ * @param string $filename
+ *   A file name.
+ *
+ * @return array
+ *   An empty array if the file name is allowed or an array containing an
+ *   error message if it's not.
+ *
+ * @see hook_file_validate()
+ */
+function file_validate_filename(FileInterface $file, $filename) {
+  $errors = [];
+
+  if ($file->getFilename() != $filename) {
+    $errors[] = t('Only file with the following name is allowed: <strong>%file-name-allowed</strong>.', ['%file-name-allowed' => $filename]);
```

Comment:
We shouldn't validate whole name. When font is uploaded, we should perform validation, just check for the term 'bold' 'book' or 'medium' in font name, in case someone uploads BOOK to BOLD, or MEDIUM etc. This was in the original requirement.

Reply:
And also in the original requirement were `Description: Upload Cachet Bold here. File name should be <strong>Cachet-Bold.ttf</strong>` and `Description: Upload Cachet Book here. File name should be <strong>Cachet-Book.ttf</strong>.` and `Description: Upload Cachet Medium here. File name should be <strong>Cachet-Medium.ttf</strong>.`. Will it be ok if user upload file named like `bold.ttf` or `book-test.ttf` or `bold-book_medium.ttf`? If yes, then the descriptions above sound a bit wrong... For me it looks like requirements conflict. This is why I implemented validation whole name.

Reply:
It's fine. We should have strict validation for the whole name. We need this just for the case when someone accidentally will upload book to bold.

Reply:
Sorry, but I didn't understand actually... So should I implement validation that will just check for the term 'bold' 'book' or 'medium' in the font filename? Or should I leave current implementation?

Reply:
What if user will rename bold font to `book.ttf`? Is it ok?

Reply:
Yes, implement only validation by terms "book", "bold" or "medium". In the validation message specify somethink like "It looks like that you're trying to upload wrong font. Font name should contain %font_style in the name".

Reply:
Ok.